custom-checks

```diff
@@ -44,24 +44,26 @@ fi
 
 shopt -s globstar
 
-directory="$1"
+directories=("$@")
 
 check_errors() {
     regex="$1"
     message="$2"
 
-    for match in $(
-        grep -n "$regex" "$directory"/**/*.vim \
-        | grep -o '^[^:]\+:[0-9]\+' \
-        | sed 's:^\./::'
-    ); do
-        RETURN_CODE=1
-        echo "$match $message"
+    for directory in "${directories[@]}"; do
+        while IFS= read -r match; do
```

Comment:
I also recommend `while read -r; do` and using the special `$REPLY`. That tends to work consistently and avoid issues with trailing and leading whitespace being removed.